packages/playground/__tests__/Header.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { Header } from '~/app/Header';
+import { render, screen, waitFor } from '@testing-library/react';
+import { AppContext, AppProvider } from '~/context';
+
+describe('Given the Header component is rendered', () => {
+  beforeEach(() => {
+    render(<Header />);
+  });
+
+  test('When it appears on the screen', () => {
+    const settingsButton = screen.getByText('Settings');
+    const logoIcon = screen.getByTestId('headerLogo');
+
+    // Then ...
+    expect(settingsButton).toBeInTheDocument();
+    expect(logoIcon).toBeInTheDocument();
+  });
+
+  test('When settings button was not clicked', () => {
+    const darkMode = screen.queryByText('Dark Mode');
+
+    // Then ...
+    expect(darkMode).toBeNull();
+  });
+
+  test('When settings button is clicked', () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    const darkMode = screen.getByText('Dark Mode');
+    const minimap = screen.getByText('Minimap');
+    const numbering = screen.getByText('Numbering');
+
+    // Then ...
+    expect(darkMode).toBeInTheDocument();
+    expect(minimap).toBeInTheDocument();
+    expect(numbering).toBeInTheDocument();
+  });
+
+  test('When settings dropdown is closed', async () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    const closeButton = screen.getByLabelText('close');
+    closeButton.click();
+
+    // Then ...
+    await waitFor(() => {
+      expect(screen.queryByText('Dark Mode')).toBeNull();
+    });
+  });
+
+  test('When settings button is clicked twice', async () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    settingsButton.click();
+
+    // Then ...
+    await waitFor(() => {
+      expect(screen.queryByText('Dark Mode')).toBeNull();
+    });
+  });
+
+  test('When defaults are unchanged', () => {
+    render(
+      <AppContext.Consumer>
+        {([state]) => {
```

Comment:
Couldn't we just have a unit test for the reducer function. And then a another one that tests if the AppContext works? Testing every action though a mocked UI context looks like an overhead to me. Or, what's the objection here?

Reply:
I think this is a good idea anyways. I'll add unit tests for the reducer.

Reply:
@thought2  I have added tests for the reducer, but I am not quite sure, if they all make sense the way I mocked the API response. Would be cool, if you could have a look.